dias/tracker.py

```diff
@@ -0,0 +1,400 @@
+
+FILETYPES = ["chimestack", "chimecal", "chimetiming"]
+
+
+class BaseModel(Model):
+    """Base Model defining the database connection."""
+
+    class Meta:
+        """The definition of the database connection."""
+
+        database = db
+
+
+class FileType(BaseModel):
+    """
+    Model definition for table of filetypes that DIAS uses.
+
+    Attributes
+    ----------
+    name : String
+        name of filetype; matches how they would be written in directory names
+    id : Integer
+        unique id for filetype in database
+    """
+
+    name = CharField(unique=True)
+
+
+class Analyzer(BaseModel):
+    """
+    Model definition for the table of analyzers.
+
+    Attributes
+    ----------
+    name : String
+        name of analyzer
+    """
+
+    name = CharField(unique=True)
+
+
+class File(BaseModel):
+    """
+    Model definition for table of files that Dias analyzers interact with.
+
+    Attributes
+    ----------
+    filepath : String
+        Absolute path for file
+    file_type_id : Integer
+        reference to filetype of that file
+    start_time : datetime.datetime
+        timestamp of earliest datapoint in file
+    end_time : datetime.datetime
+        timestamp of latest datapoint in file
+    exists : Boolean
+        does the file still exist at filepath
+    """
+
+    filepath = CharField(unique=True)
+    file_type_id = ForeignKeyField(FileType, backref="files")
+    start_time = DateTimeField()
+    end_time = DateTimeField()
+    exists = BooleanField()
+
+
+class Processed(BaseModel):
+    """
+    Model for table of analyzer/file pairs which tracks which files have already been processed by the analyzers.
+
+    Attributes
+    ----------
+    file_id : Integer
+        Refers to a file in a file table
+    analyzer_id : Integer
+        Refers to an analyzer in the analyzer table
+    """
+
+    file_id = ForeignKeyField(File, backref="processed")
+    analyzer_id = ForeignKeyField(Analyzer, backref="processed")
+
+    class Meta:
+        """Meta for the Model."""
+
+        database = db
+
+        # file/analyzer pairs must be unique
+        indexes = ((("file_id", "analyzer_id"), True),)
+
+
+class Tracker:
+    """
+    Utility that keeps track of processed files for DIAS analyzers.
+
+    Interfaces with an SQLite database to assess which files have/haven't been already processed by various analyzers.
+    Creates the database, if it does not exist, and keeps it up-to-date with the current state of the staging directory.
+
+    Attributes
+    ----------
+    path : String
+        the location of the staging directory that the user is interested in
+
+    Methods
+    -------
+    __init__
+    create_tables
+    _get_filetypes
+    insert_files
+    add_analyzer_ine
+    remove_files
+    update_file_table
+    new_files
+    register_done
+    """
+
+    def __init__(
+        self,
+        path="/mnt/gong/staging",
+        file_tracking_db="/mnt/gong/staging/file_index.db",
```

Comment:
I don't think the file should get stored there. Files in staging should be generated by acquisition processes only and should generally be ephemeral. It's not clear that storing a sqlite db on a network FS is a great idea for performance either. Maybe @ketiltrout has thoughts about where it should go? I appreciate that the *actual* location for this is in ch_ansible, but it's currently the same as the default here.

Reply:
The current databases that other DIAS analyzers reference are also on gong! So if we change this one, we should possibly move all of those. 

Maybe we place them in the same place as `dias.conf` (`/etc/dias`)?

Reply:
Usually for this kind of thing `/var/run/dias/` would be the appropriate location, but I think it needs to get backed up from wherever it is stored, and I think that is. `/mnt/gong/run/dias/` could be an option.

Reply:
`/var/cache/dias/` would be better than `/var/run/`, which is meant for process-specific files.

Reply:
Also, please, not `/mnt/gong/run/dias`.  Just use `/mnt/gong/dias/` if you must.

Reply:
(i.e. `/mnt/gong/dias/file_index.db` would be fine)

Reply:
I will use `/var/cache/dias/`!

Reply:
Nevermind! Per a slack discussion, I am going to use /var/lib/dias with a copy to gong in a cronjob.

We want to minimise frequent connection through a NFS for performance reasons. I am also going to update the dias ch_ansible PR.